### PR TITLE
SoftMax improvments:

### PR DESCRIPTION
**Pull request by moshehoori**
1. FCD version is now opitimized using VLM and unroll
2. both versions now support gaudi2
3. formula changed for both FCD and non FCD versions to gain numerical stability

### CODE REVIEW

**Review by moshehoori — COMMENTED** [review]
@tzachicohen , please review my code.
Changes added : 
1. softmax FCD is now optimized
2. gaudi 2 now supported in both FCD/nonFCD
3. softmax formula changed, to achieve numerical stability

File: kernels/gaudi/softmax_non_fcd_bf16.c [hunk 1]
```diff
@@ -68,14 +74,30 @@ void main(
             {
                 ifmCoords[height] = h;
 
-                sum = zero_bf16;
 
+                max = neg_inf_bf16;
+                for (int w = widthStart; w < widthEnd; w += widthStep)
+                {
+                    ifmCoords[width] = w;
+
```
Comment (tzachicohen) [inline kernels/gaudi/softmax_non_fcd_bf16.c:82]: remove empty line

File: kernels/gaudi/softmax_non_fcd_bf16.c [hunk 2]
```diff
@@ -68,14 +74,30 @@ void main(
...
+                    // load input pixel
+                    x = v_bf16_ld_tnsr_b(ifmCoords, ifm);
+                    // Move -inf for out of bound co-ordinates
+                    bool256 pred = from_bool128(v_u16_cmp_geq_b(d + V_LANE_ID_16, (unsigned)depthEnd, 0, to_bool128((bool256){0})));
+                    y = v_bf16_mov_vb(neg_inf_bf16, 0, x, to_bool128(pred), 0);
+
```
Comment (tzachicohen) [inline kernels/gaudi/softmax_non_fcd_bf16.c:88]: remove empty line

File: src/gaudi2_src/softmax_bf16_gaudi2.cpp
Comment (tzachicohen) [inline src/gaudi2_src/softmax_bf16_gaudi2.cpp:1]: We usually have one glue code for all devices , otherwise it is a horrible code replication.

Comment (moshehoori) [reply src/gaudi2_src/softmax_bf16_gaudi2.cpp:1]: @tzachicohen , it looks like currently in this repo all other kernel's glue code are split between gaudi1 and gaudi2 (not sure what is the reason). should I unite all of them?

**Review by tzachicohen — APPROVED** [review]

**Pull request merged**